### PR TITLE
CDAP-5322 Resolve schema for existing filesets using a heuristic

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA - https://issues.cask.co/browse/CDAP-5322
Build - http://builds.cask.co/browse/CDAP-DUT3851-3

### CODE REVIEW

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Test DatasetSystemMetadataWriter
+ */
+public class DatasetSystemMetadataWriterTest {
+  @Test
+  public void testFilesetSchema() throws Exception {
+    DatasetProperties filesetAvroTableProps =
+      FileSetProperties.builder()
+        .setTableProperty("avro.schema.literal", "avro-table-schema")
```
Comment (bdmogal) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java]: should we make these constants package-private and use them in this test?

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        .add(DatasetProperties.SCHEMA, "avro-schema")
+        .build();
+    metadataWriter =
+      new DatasetSystemMetadataWriter(new NoOpMetadataStore(), Id.DatasetInstance.from("ns1", "avro1"),
+                                      filesetAvroTableProps, null, null, null);
+    Assert.assertEquals("avro-schema", metadataWriter.getSchemaToAdd());
+
+    DatasetProperties filesetAvroOutputProps =
```
Comment (bdmogal) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java]: nit: can these verifications be easily extracted to methods? seems like a similar code.

**Comment by bdmogal** [comment]
just a couple comments. rest LGTM

**Comment by poornachandra** [comment]
@bdmogal Addressed comments, please take another look.

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    DatasetProperties filesetParquetProps =
+      FileSetProperties.builder()
+        .setOutputProperty(DatasetSystemMetadataWriter.FILESET_PARQUET_SCHEMA_OUTPUT_KEY, "parquet-output-schema")
+        .build();
+    assertDatasetSchema("parquet-output-schema", filesetParquetProps);
+
+    // When SCHEMA property is present, it should override
+    filesetParquetProps =
+      FileSetProperties.builder()
+        .setOutputProperty(DatasetSystemMetadataWriter.FILESET_PARQUET_SCHEMA_OUTPUT_KEY, "parquet-schema")
```
Comment (bdmogal) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java]: this should be called ``"parquet-output-schema"`` so we're sure that the overriding happens.

**Comment by bdmogal** [comment]
1 final comment. No need to wait for build though. LGTM after the fix pending local test/checkstyle run.

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java (hunk 3 above)
Comment (poornachandra) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/system/DatasetSystemMetadataWriterTest.java]: Thanks, will fix it.

**Comment by poornachandra** [comment]
Build is green, merging.

**Pull request merged**